### PR TITLE
Fix logic for forwarding birth registration to mosip

### LINKED ISSUE #12731: Record does not go to the "Pending external validation" workqueue for a birth declaration when only informant identity is authenticated via e-Signet and no parent details are available
**Bug description:**
When a birth declaration is submitted where the child has no father or mother details available and only informant information is provided, and the informant's identity has been authenticated via e-Signet — the record is not routed to the external validation workqueue and there is no log recorded as `Waiting for external validation` for this in the audit tab

**Steps to reproduce:**
1. Log in as a Registrar and initiate a new birth declaration.
2. Enter the child's details
3. Select an informant who is neither the father nor the mother.
4. Provide the informant's details and verify their identity using e-Signet authentication.
5. Mark both the father's and mother's details as unavailable.
6. Submit the declaration.

**Actual result:**
- The record is not routed to the external validation workqueue after approval and registration. The record bypasses the external validation step entirely despite the informant having completed e-Signet authentication.
- No "Waiting for external validation" log entry is present in the Audit tab after approval and registration. The audit trail does not reflect the external validation step, despite the informant having been authenticated through e-Signet

**Expected result:**
- After registration, the record should be routed to the external validation workqueue, reflecting that the informant's identity was authenticated via e-Signet and the record requires external validation.
- A "Waiting for external validation" log entry should appear in the Audit tab after the record is approved and registered, reflecting that the informant's identity has gone through external e-Signet validation during the process.

**Screenshot/Recording:**

- https://drive.google.com/file/d/10RMIqbf_Jto6KmGMKX8AH_8NdSfz6wHa/view?usp=drive_link

**Tested on:**
https://register.farajaland-demo-staging.opencrvs.org/

### LINKED ISSUE #12813: Birth records getting stuck in pending external validation workqueue when authenticated data is provided as Registrar
**Bug Title:**
Records getting stuck in pending external validation workqueue when authenticated data is provided as Registrar

**Steps to produce:**
1. Declare birth event with authenticated identity for informant as any user
2. Try to register that events record as registrar
3. Observe the behavior

**Actual result:**
Records are getting stuck in pending external validation workqueue

**Expected result:**
Records should be found in pending certification workqueue

**Screen recording:**

https://github.com/user-attachments/assets/df142f1f-0075-4eaf-b79e-39d3bbe7c3fe

**Tested on:**
https://login.farajaland-demo-staging.opencrvs.org/?lang=en

### PR DESCRIPTION
**Pull request by cibelius**
## Description

Now it takes in to account both requirements:
Resolves https://github.com/opencrvs/opencrvs-core/issues/12813
Resolves https://github.com/opencrvs/opencrvs-core/issues/12731

Related discussion here: https://opencrvsworkspace.slack.com/archives/C07ASGRUW04/p1780490663861239

## Checklist

- [x] I have linked the correct Github issue under "Development"
- [x] I have tested the changes locally, and written appropriate tests
- [x] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [x] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: src/events/mosip.ts [hunk 1]
```diff
@@ -391,10 +398,12 @@ export function shouldForwardBirthRegistrationToMosip(
     declaration['mother.verified'] !== 'verified' &&
     declaration['mother.verified'] !== 'authenticated' &&
     declaration['father.verified'] !== 'verified' &&
-    declaration['father.verified'] !== 'authenticated'
+    declaration['father.verified'] !== 'authenticated' &&
+    declaration['informant.verified'] !== 'verified' &&
+    declaration['informant.verified'] !== 'authenticated'
```
Comment (cibelius) [inline src/events/mosip.ts:403]: The actual intended fix

**Review by jamil314 — APPROVED** [review]
